src/wp-includes/blocks.php

```diff
@@ -151,10 +151,12 @@ function register_block_style_handle( $metadata, $field_name ) {
 		$style_uri  = includes_url( 'blocks/' . str_replace( 'core/', '', $metadata['name'] ) . "/style$suffix.css" );
 	}
 
-	$style_handle = generate_block_asset_handle( $metadata['name'], $field_name );
-	$block_dir    = dirname( $metadata['file'] );
-	$style_file   = realpath( "$block_dir/$style_path" );
-	$version      = file_exists( $style_file ) ? filemtime( $style_file ) : false;
+	$style_handle   = generate_block_asset_handle( $metadata['name'], $field_name );
+	$block_dir      = dirname( $metadata['file'] );
+	$style_file     = realpath( "$block_dir/$style_path" );
+	$has_style_file = false !== $style_file;
+	$version        = ! $is_core_block && $has_style_file ? filemtime( $style_file ) : false;
```

Comment:
Why is `filemtime` needed? Why can't the version be something that the block defines? Perhaps if it doesn't exist we use the WP version?

Reply:
@aaronjorbin [Trac ticket 53507](https://core.trac.wordpress.org/ticket/53507) opened to replace `filemtime` using the version in the metadata.